PowerPointLabs/PowerPointLabs/SpeechEngine/TextToSpeech.cs

```diff
@@ -59,11 +68,106 @@ public static void SpeakString(String textToSpeak)
             PromptToAudio.Speak(builder);
         }
 
+        private static void SaveStringToWaveFileWithHumanVoice(string textToSave, string filePath)
+        {
+            string accessToken;
+            string textToSpeak = GetHumanSpeakNotesForText(textToSave);
+
+            try
+            {
+                Authentication auth = Authentication.GetInstance();
+                accessToken = auth.GetAccessToken();
+                Logger.Log("Token: " + accessToken);
+            }
+            catch (Exception ex)
+            {
+                Logger.Log("Failed authentication.");
+                Logger.Log(ex.ToString());
+                Logger.Log(ex.Message);
+                return;
+            }
+            string requestUri = UserAccount.GetInstance().GetUri();
+            var cortana = new Synthesize();
+
+            cortana.OnAudioAvailable += SaveAudioToWaveFile;
+            cortana.OnError += ErrorHandler;
+
+            // Reuse Synthesize object to minimize latency
+            cortana.Speak(CancellationToken.None, new Synthesize.InputOptions()

```

Comment:
The default values are not currently used in code because attributes in InputOptions are always reset when issuing request.